test/e2e/storage/persistent_volumes-disruptive.go

```diff
@@ -105,44 +108,38 @@ var _ = framework.KubeDescribe("PersistentVolumes [Volume][Disruptive][Flaky]",
 			Expect(clientNodeIP).NotTo(BeEmpty())
 		}
 	})
-
 	AfterEach(func() {
 		framework.DeletePodWithWait(f, c, nfsServerPod)
 	})
-
 	Context("when kubelet restarts", func() {
 
 		var (
 			clientPod *v1.Pod
 			pv        *v1.PersistentVolume
 			pvc       *v1.PersistentVolumeClaim
 		)
-
 		BeforeEach(func() {
 			framework.Logf("Initializing test spec")
 			clientPod, pv, pvc = initTestCase(f, c, nfsPVconfig, pvcConfig, ns, clientNode.Name)
 		})
-
 		AfterEach(func() {
 			framework.Logf("Tearing down test spec")
-			tearDownTestCase(c, f, ns, clientPod, pvc, pv)
+			tearDownTestCase(c, f, ns, clientPod, nfsServerPod, pvc, pv)
```

Comment:
The nfs-server pod is deleted in the outer `AfterEach` and here. Is that intentional?